examples/src/Components/FileSelect.tsx

```diff
@@ -16,57 +13,58 @@ interface FileSelectionProps {
 
 const FileSelection = ({
     selectedFile,
+    conversionFileName,
     onFileSelect,
     loadSmoldynFile,
     clearFile,
     loadSmoldynPreConfiguredSim,
     setRabbitCount,
 }: FileSelectionProps): JSX.Element => {
+    const [selectValue, setSelectValue] = React.useState("");
+
     useEffect(() => {
         const urlParams = new URLSearchParams(window.location.search);
         if (urlParams.has("file")) {
             const queryStringFile = urlParams.get("file") || "";
             onFileSelect(queryStringFile);
         }
-    }, [onFileSelect]);
+    }, []);
+
+    useEffect(() => {
+        if (conversionFileName !== "") {
+            setSelectValue("Awaiting file conversion...");
+        } else {
+            setSelectValue(selectedFile);
+        }
+    }, [selectedFile, conversionFileName]);
 
     const handleFileSelect = useCallback(
         (file: string) => {
-            if (
-                selectedFile === AWAITING_SMOLDYN_SIM_RUN ||
-                selectedFile === AWAITING_CONVERSION
-            ) {
+            if (conversionFileName !== "") {
                 return;
             }
             onFileSelect(file);
         },
-        [selectedFile, onFileSelect]
+        [selectedFile, onFileSelect, conversionFileName]
     );
 
     const notInList =
-        selectedFile &&
-        selectedFile !== AWAITING_SMOLDYN_SIM_RUN &&
-        selectedFile !== AWAITING_CONVERSION &&
-        !TRAJECTORY_OPTIONS.some((t) => t.id === selectedFile);
+        selectedFile && !TRAJECTORY_OPTIONS.some((t) => t.id === selectedFile);
 
     return (
         <div className={"ui-container"}>
             <select
-                value={selectedFile}
+                value={selectValue}
                 onChange={(e) => handleFileSelect(e.target.value as string)}
                 style={{ maxWidth: 200 }}
+                disabled={conversionFileName !== ""}
```

Comment:
Higher-level feedback, but I feel like making a boolean constant (`isAwaitingFileConversion`) based on this would make it easier to understand the code. It's not immediately obvious to me that the conversion filename being non-empty means something?

It could also be a sign that `conversionFileName` could be `pendingConversionFileName` to indicate that it's a temporary value.